src/lib/coveo/coveoCustom.ts

```diff
@@ -0,0 +1,212 @@
+import { Reflection, ReflectionFlag, CommentTag } from "../models";
+import { parseComment } from "../converter/factories/comment";
+import { Converter } from "../converter";
+
+export interface ICoveoDisplayTemplateFunction {
+  (rawValue?: string, result?: Reflection): string;
+}
+
+export class CoveoCustomTag {
+  public displayName: string;
+  public displayTemplate: ICoveoDisplayTemplateFunction;
+  public rawName?: string;
+  public rawValue?: string;
+
+  constructor(displayName: string, displayTemplate: ICoveoDisplayTemplateFunction, rawName?: string, rawValue?: string) {
+    this.displayName = displayName;
+    this.displayTemplate = displayTemplate;
+    this.rawName = rawName;
+    this.rawValue = rawValue;
+  }
+}
+
+export interface IParsedCoveoBuildOptionArgs {
+  required?: CoveoCustomTag,
+  defaultValue?: CoveoCustomTag,
+  depend?: CoveoCustomTag,
+  min?: CoveoCustomTag,
+  max?: CoveoCustomTag,
+  float?: CoveoCustomTag
+}
+
+export class CoveoCustom {
```

Comment:
There's quite a bit of logic here, I wonder if we should start adding unit tests to this project 😬

Reply:
It would probably be a good idea, yes :)